src/counters/CountersDDR3.h

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2012-2020, TU Delft
+ * Copyright (c) 2012-2020, TU Eindhoven
+ * Copyright (c) 2012-2020, TU Kaiserslautern
+ * Copyright (c) 2012-2020, Fraunhofer IESE
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ * notice, this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright
+ * notice, this list of conditions and the following disclaimer in the
+ * documentation and/or other materials provided with the distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+ * IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+ * TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+ * PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
+ * TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * Authors: Karthik Chandrasekar,
+ *          Matthias Jung,
+ *          Omar Naji,
+ *          Sven Goossens,
+ *          Éder F. Zulian
+ *          Subash Kannoth
+ *          Felipe S. Prado
+ *          Luiza Correa
+ */
+#ifndef COUNTERSDDR3_H
+#define COUNTERSDDR3_H
+
+#include <stdint.h>
+
+#include <vector>
+#include <iostream>
+#include <deque>
+#include <string>
+
+#include "MemCommand.h"
+#include "./memspec/MemSpecDDR3.h"
+#include "Counters.h"
+
+namespace DRAMPower {
+class CountersDDR3 final : public Counters {
+ public:
+
+  CountersDDR3(MemSpecDDR3& memspec);
+  MemSpecDDR3 memSpec;
+  void getCommands(std::vector<MemCommand>& list,
+                             bool lastupdate,
+                             int64_t timestamp);
+
+
+  // Handlers for commands that are getting processed
+
+  void handleRef(    unsigned bank, int64_t timestamp);
```

Comment:
we think that all overwritten functions should have a virtual and override specifier.